ArduPlane/ArduPlane.cpp

```diff
@@ -968,6 +969,81 @@ void Plane::update_flight_stage(void)
     airspeed.set_EAS2TAS(barometer.get_EAS2TAS());
 }
 
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell and Samuel Tabor
+*/
+void Plane::update_soaring() {
+    
+    if (!soaring_controller.is_active()) {
+        return;
```

Comment:
Because of the way the soaring controller's logic currently works, there is no need to reset/do anything special for the enable/disable to work correctly. I'm more worried that many of soaring controller's member variables are used before they are initialized (e.g., _prev_vario_update_time and _last_alt). Again, we currently get away with this because of the specifics of soaring controller's operation, but changing this logic even slightly may suddenly cause bizarre behavior. @samuelctabor, do you want to pitch in here? What's the best way to reliably initialize SoaringController's members? Do they need to be reset in a specific way when SoaringController is deactivated in mid-flight?